### ISSUE TITLE
"sidebar" listed as orphan

### ISSUE THREAD

**Issue by phallobst** [i-net-software/dokuwiki-plugin-multiorphan#42]
The nature of the "sidebar" page does not require any referencing link in order to be used, so it should be excluded from the orphan list.

**Comment by gamma** [comment]
Yes, but there are some more files that will have to be excluded afaik.

**Comment by sivann** [comment]
also, all files referenced by sidebar are listed as orphans, but they are not.

**Comment by schniepp** [comment]
@gamma: which files did you have in mind? I think it would mainly be all the pages named "sidebar" (sidebar.txt). These page names could simply be excluded from the search. The only downside would be if someone has named a page "sidebar" without having the sidebar plugin installed. A simple solution would be a checkbox, that sidebar used could check. A more sophisticated solution would be to detect the presence of an enabled plugin "sidebar", which then automatically enables the exclusion of pages names "sidebar".

**Comment by rgleason** [comment]
I think this would help a lot.

**Comment by gamma** [comment]
Sidebar is just once thing. E.g. were using "toc" files heavily - so I'd exclude them too, but that is not a thing for many other users. I'm think a global option filled with default excludes would fit the cases better. "sidebar" could be a default one.

**Comment by schniepp** [comment]
@gamma: sure that sounds like an amazing solution!

**Comment by rgleason** [comment]
To confirm, we use Sidebar  TOC  and indexmenu extensively.

**Comment by gamma** [comment]
> To confirm, we use Sidebar TOC and indexmenu extensively.

You're referring plugins? I meant page names. Anyway, it can be customised with a good default.

**Comment by rgleason** [comment]
Yes we use the sidebar and a TOC,   along with the indexmenu plugin.

**Comment by gamma** [comment]
BTW: There is an exclude pattern in the multiorphan page already.

**Comment by rgleason** [comment]
Good point

**Comment by schniepp** [comment]
> BTW: There is an exclude pattern in the multiorphan page already.

I saw that... but unfortunately, I could not find any documentation how that works. It wasn't even clear to me whether this was an exclude or include pattern. But thanks for pointing that out! Perhaps, I should just play with it.

**Comment by rgleason** [comment]
@schiepp  Did you figure out what to enter?

**Comment by schniepp** [comment]
@rgleason : as far as I can tell, the field `RegEx to filter full ID:` is a positive list (include pattern). When I insert simply `sidebar`, it only queries my sidebar pages. As far as I know, there is no NOT operator in regular expressions, so I don't see an obvious way of turning this into a negative list (exclusions), which is what would be needed here. (To exclude all sidebar pages, etc.)

**Comment by asheenlevrai** [comment]
Hi :)

Can anyone explain how _RegEx to filter full ID_ works and how it can be used to filter results?

Thanks a lot

-a-

**Comment by rgleason** [comment]
For [OpenCPN Wiki  ](https://opencpn.org/wiki/dokuwiki/doku.php?id=opencpn:manual_basic)
we use blank pages to control how sub namespaces appear.  These get in the way sometimes and I want to exclude them.
So using the RegEx line.

RegEx: This eliminates all the *:blank pages under opencpn:manual_basic

   ^(?!.*opencpn:manual_basic:.*:blank).*$

See this Editor Manual page  https://opencpn.org/wiki/dokuwiki/doku.php?id=opencpn:edit_user_manual:administrator:multiorphan_checker
